### ISSUE TITLE
Using standalone HttpProvider does not work.

### ISSUE THREAD

**Issue by ksikorsk** [microsoftgraph/msgraph-sdk-dotnet#447]
When using the HttpProvider with a custom message handler, the message handler never gets invoked.  Instead the HttpProvider throws: `ServiceException : Code: invalidRequest`

I've attached a unit test below.  the `MockMessageHandler` just validates the request and returns with a `202 Accepted`

### Expected behavior

```
Assert.AreEqual(1, messageHandlerTriggered);
Assert.AreEqual(1, authProviderTriggered);
```

### Actual behavior

```
Message: Microsoft.Graph.ServiceException : Code: invalidRequest
Message: Authentication provider is required before sending a request.
```

### Steps to reproduce the behavior

```cs
var tenant = Guid.NewGuid().ToString();
var authProviderTriggered = 0;
var authProvider = new DelegateAuthenticationProvider(message =>
{
    authProviderTriggered++;
    message.Headers.Authorization = new AuthenticationHeaderValue("bearer", tenant);
    return Task.CompletedTask;
});

var url = new Uri("https://Test");
var messageHandlerTriggered = 0;
var messageHandler = new MockMessageHandler(message =>
{
    messageHandlerTriggered++;
    Assert.AreEqual(tenant, message.Headers?.Authorization?.Parameter);
    Assert.AreEqual(url, message.RequestUri);
});

var authHandler = new AuthenticationHandler(authProvider, messageHandler);

var httpProvider = new HttpProvider(
    authHandler,
    true,
    new CommsSerializer(pretty: true));

var requestToSend = new HttpRequestMessage
{
    Method = HttpMethod.Post,
    RequestUri = url,
};

var response = await httpProvider.SendAsync(requestToSend).ConfigureAwait(false);
Assert.IsTrue(response.IsSuccessStatusCode);
Assert.AreEqual(1, messageHandlerTriggered);
Assert.AreEqual(1, authProviderTriggered);
```

### Root Cause

When creating the HttpHandler you specify a null IAuthenticationProvider:
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/492188c7137c57b61e59559294347c3b0bf89f0c/src/Microsoft.Graph.Core/Requests/HttpProvider.cs#L65

Which created default handlers with an `new AuthenticationHandler(null)`:
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/492188c7137c57b61e59559294347c3b0bf89f0c/src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs#L128-L138

So when I do `await httpProvider.SendAsync(requestToSend);` the first AuthenticationHandler that gets hit is the one with a null IAuthenticationProvider.

**Comment by darrelmiller** [comment]
@pwombwa Was this the issue we fixed in 1.14.1?

**Comment by ksikorsk** [comment]
I should mention this is version 1.15.0-prerelease.1

**Comment by darrelmiller** [comment]
@ksboora Yeah, I think we made a bad assumption that custom HttpProvider would actually be a different class.  You are providing a "custom" HttpProvider that is actually the same class.  We need some way to indicate that a HttpProvider was provided externallly, so as to revert back to the old auth pattern.

You could temporary workaround this issue by creating a MyHttpProvider that just derives from HttpProvider, but we will get this fixed in the 1.15 preview.

**Comment by pwombwa** [comment]
@ksikorsk, This issue has been fixed as part of [Microsoft.Graph.Core 1.15.0-preview.2](https://www.nuget.org/packages/Microsoft.Graph.Core/1.15.0-preview.2). Give it a try and let us know if it addresses your scenario(s).
